datastreams/context.go

```diff
@@ -0,0 +1,104 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package datastreams
```

Comment:
Have you considered putting the public types and functions in a `datastreams/internal` package instead of exposing everything to users in the current `datastreams` package ? By having an internal package you'll be able to change it freely in the future without any backward-compatibility restrictions. This also a common practice in dd-trace-go, see `ddtrace/internal`,  `profiler/internal` and `contrib/internal`.